### ISSUE TITLE
IllegalStateException: Could not self-attach to current VM using external process on Open JDK 11.0.1

### ISSUE THREAD

**Issue by Sam-Kruglov** [mockk/mockk#254]
We have main code in Java and tests in Kotlin. We build it and run the tests inside Docker container. **All works locally** but it does not work in our build environment.
We have a Jenkins master running on AWS. It calls a slave node (also on AWS) to run the build. The build fails because of this library with the following error: 
```
[WARNING] Corrupted STDOUT by directly writing to native stream in forked JVM 1. See FAQ web page and the dump file /home/mbuilder/workspace/....dumpstream
java.lang.ExceptionInInitializerError
Caused by: java.lang.IllegalStateException: Could not self-attach to current VM using external process
java.lang.NoClassDefFoundError: Could not initialize class io.mockk.impl.JvmMockKGateway
```
The dumpstream file is attached [2019-02-12T15-01-39_092-jvmRun1.dumpstream](https://github.com/mockk/mockk/files/2860170/2019-02-12T15-01-39_092-jvmRun1.dumpstream.log)(had to add `.log` extention for GitHub). 
Also, note that `OurClassTest` mentioned in the dumpstream is only 283 lines long, so there is no line 290 that is referenced there.

**Comment by Sam-Kruglov** [comment]
mockk 1.9
surefire 2.22.1
junit 5.3.2
open jdk 11.0.1

**Comment by Sam-Kruglov** [comment]
I suppose that should be somehow connected to the fact that this is happening inside AWS. Maybe it is close to this https://hibernate.atlassian.net/browse/OGM-1461

**Comment by oleksiyp** [comment]
May be it is related to test directory. https://github.com/mockk/mockk/issues/239. Try 1.9.1

**Comment by Sam-Kruglov** [comment]
@oleksiyp hey, just tried it, didn't help :(

```
--- maven-dependency-plugin:3.1.1:tree (default-cli) @ card-service ---
[INFO] com.monese.card:card-service:pom:0.1.11
[INFO] +- org.projectlombok:lombok:jar:1.18.4:provided
[INFO] +- org.apache.commons:commons-lang3:jar:3.8.1:compile
[INFO] +- org.jetbrains.kotlin:kotlin-stdlib-jdk8:jar:1.3.11:compile
[INFO] |  +- org.jetbrains.kotlin:kotlin-stdlib:jar:1.3.11:compile
[INFO] |  |  +- org.jetbrains.kotlin:kotlin-stdlib-common:jar:1.3.11:compile
[INFO] |  |  \- org.jetbrains:annotations:jar:13.0:compile
[INFO] |  \- org.jetbrains.kotlin:kotlin-stdlib-jdk7:jar:1.3.11:compile
[INFO] +- org.junit.jupiter:junit-jupiter-api:jar:5.3.2:test
[INFO] |  +- org.apiguardian:apiguardian-api:jar:1.0.0:test
[INFO] |  +- org.opentest4j:opentest4j:jar:1.1.1:test
[INFO] |  \- org.junit.platform:junit-platform-commons:jar:1.3.2:test
[INFO] +- org.junit.jupiter:junit-jupiter-engine:jar:5.3.2:test
[INFO] |  \- org.junit.platform:junit-platform-engine:jar:1.3.2:test
[INFO] +- org.junit.jupiter:junit-jupiter-params:jar:5.3.2:test
[INFO] +- io.mockk:mockk:jar:1.9.1:test
[INFO] |  +- io.mockk:mockk-common:jar:1.9.1:test
[INFO] |  |  \- io.mockk:mockk-dsl:jar:1.9.1:test
[INFO] |  +- io.mockk:mockk-dsl-jvm:jar:1.9.1:test
[INFO] |  +- io.mockk:mockk-agent-jvm:jar:1.9.1:test
[INFO] |  |  +- io.mockk:mockk-agent-api:jar:1.9.1:test
[INFO] |  |  +- io.mockk:mockk-agent-common:jar:1.9.1:test
[INFO] |  |  +- org.objenesis:objenesis:jar:2.6:test
[INFO] |  |  +- net.bytebuddy:byte-buddy:jar:1.8.8:test
[INFO] |  |  \- net.bytebuddy:byte-buddy-agent:jar:1.8.8:test
[INFO] |  \- org.jetbrains.kotlin:kotlin-reflect:jar:1.3.11:test
[INFO] +- org.springframework.boot:spring-boot-starter-test:jar:2.1.1.RELEASE:test
[INFO] |  +- org.springframework.boot:spring-boot-starter:jar:2.1.1.RELEASE:test
[INFO] |  |  +- org.springframework.boot:spring-boot:jar:2.1.1.RELEASE:test
[INFO] |  |  |  \- org.springframework:spring-context:jar:5.1.3.RELEASE:test
[INFO] |  |  |     +- org.springframework:spring-aop:jar:5.1.3.RELEASE:test
[INFO] |  |  |     +- org.springframework:spring-beans:jar:5.1.3.RELEASE:test
[INFO] |  |  |     \- org.springframework:spring-expression:jar:5.1.3.RELEASE:test
[INFO] |  |  +- org.springframework.boot:spring-boot-autoconfigure:jar:2.1.1.RELEASE:test
[INFO] |  |  +- org.springframework.boot:spring-boot-starter-logging:jar:2.1.1.RELEASE:test
[INFO] |  |  |  +- ch.qos.logback:logback-classic:jar:1.2.3:test
[INFO] |  |  |  |  \- ch.qos.logback:logback-core:jar:1.2.3:test
[INFO] |  |  |  +- org.apache.logging.log4j:log4j-to-slf4j:jar:2.11.1:test
[INFO] |  |  |  |  \- org.apache.logging.log4j:log4j-api:jar:2.11.1:test
[INFO] |  |  |  \- org.slf4j:jul-to-slf4j:jar:1.7.25:test
[INFO] |  |  +- javax.annotation:javax.annotation-api:jar:1.3.2:test
[INFO] |  |  \- org.yaml:snakeyaml:jar:1.23:test
[INFO] |  +- org.springframework.boot:spring-boot-test:jar:2.1.1.RELEASE:test
[INFO] |  +- org.springframework.boot:spring-boot-test-autoconfigure:jar:2.1.1.RELEASE:test
[INFO] |  +- com.jayway.jsonpath:json-path:jar:2.4.0:test
[INFO] |  |  +- net.minidev:json-smart:jar:2.3:test
[INFO] |  |  |  \- net.minidev:accessors-smart:jar:1.2:test
[INFO] |  |  |     \- org.ow2.asm:asm:jar:5.0.4:test
[INFO] |  |  \- org.slf4j:slf4j-api:jar:1.7.25:test
[INFO] |  +- junit:junit:jar:4.12:test
[INFO] |  +- org.assertj:assertj-core:jar:3.11.1:test
[INFO] |  +- org.mockito:mockito-core:jar:2.23.0:test
[INFO] |  +- org.hamcrest:hamcrest-core:jar:1.3:test
[INFO] |  +- org.hamcrest:hamcrest-library:jar:1.3:test
[INFO] |  +- org.skyscreamer:jsonassert:jar:1.5.0:test
[INFO] |  |  \- com.vaadin.external.google:android-json:jar:0.0.20131108.vaadin1:test
[INFO] |  +- org.springframework:spring-core:jar:5.1.3.RELEASE:test
[INFO] |  |  \- org.springframework:spring-jcl:jar:5.1.3.RELEASE:test
[INFO] |  +- org.springframework:spring-test:jar:5.1.3.RELEASE:test
[INFO] |  \- org.xmlunit:xmlunit-core:jar:2.6.2:test
[INFO] |     \- javax.xml.bind:jaxb-api:jar:2.3.1:test
[INFO] |        \- javax.activation:javax.activation-api:jar:1.2.0:test
[INFO] +- org.testcontainers:postgresql:jar:1.10.3:test
[INFO] |  \- org.testcontainers:jdbc:jar:1.10.3:test
[INFO] |     \- org.testcontainers:database-commons:jar:1.10.3:test
[INFO] |        \- org.testcontainers:testcontainers:jar:1.10.3:test
[INFO] |           +- org.apache.commons:commons-compress:jar:1.18:test
[INFO] |           +- org.rnorth.duct-tape:duct-tape:jar:1.0.7:test
[INFO] |           +- org.rnorth.visible-assertions:visible-assertions:jar:2.1.1:test
[INFO] |           |  \- net.java.dev.jna:jna:jar:4.5.2:test
[INFO] |           +- org.rnorth:tcp-unix-socket-proxy:jar:1.0.2:test
[INFO] |           |  +- com.kohlschutter.junixsocket:junixsocket-native-common:jar:2.0.4:test
[INFO] |           |  |  \- org.scijava:native-lib-loader:jar:2.0.2:test
[INFO] |           |  \- com.kohlschutter.junixsocket:junixsocket-common:jar:2.0.4:test
[INFO] |           \- net.java.dev.jna:jna-platform:jar:4.5.2:test
[INFO] \- org.jetbrains.kotlin:kotlin-test:jar:1.3.11:test
[INFO]    \- org.jetbrains.kotlin:kotlin-test-common:jar:1.3.11:test
```

**Comment by oleksiyp** [comment]
Maybe it is because you have JVM and not JDK: https://github.com/raphw/byte-buddy/issues/374

**Comment by Sam-Kruglov** [comment]
@oleksiyp I'm not entirely sure what this issue describes but our java definitely points to a JDK.
Inside the Dockerfile:
```
RUN tar zxf /openjdk-11.0.1_linux-x64_bin.tar.gz -C /usr/local \
    && rm /openjdk-11.0.1_linux-x64_bin.tar.gz

RUN alternatives --install /usr/bin/java java /usr/local/jdk-11.0.1/bin/java 1 \
    && alternatives --install /usr/bin/javac javac /usr/local/jdk-11.0.1/bin/javac 1 \
    && yum clean all

ENV JAVA_HOME=/usr/local/jdk-11.0.1
```

**Comment by deeyes24** [comment]
Hi @Sam-Kruglov : I have looked at the documentation for ByteBuddy's Agent install. 
http://bytebuddy.net/javadoc/1.5.7/net/bytebuddy/agent/ByteBuddyAgent.html#install-- 
As per above link, it does not support jdk-11 

From your observations, you said the builds passed on local. Can you share which version of JDK you have locally?

I looked at the mockito, bytebuddy for similar issues and found this one here https://github.com/mockito/mockito/issues/1298

**Comment by deeyes24** [comment]
Hi @oleksiyp , As per this ByteBuddy's documentation, https://bytebuddy.net/javadoc/1.9.9/index.html 
Could the Issue be because of JDK-11.0.1 and not finding jdk.attach module
`public static Instrumentation install()
Installs an agent on the currently running Java virtual machine. Unfortunately, this does not always work. The runtime installation of a Java agent is supported for:

JVM version 9+: For Java VM of at least version 9, the attachment API was merged into a Jigsaw module and the runtime installation is always possible.
OpenJDK / Oracle JDK / IBM J9 versions 8-: The installation for HotSpot is only possible when bundled with a JDK and requires the jdk.attach module since Java 9, if this package is placed on the module path. It is not possible for runtime-only installations of HotSpot or J9 for these versions.
If an agent cannot be installed, an IllegalStateException is thrown.

Important: This is a rather computation-heavy operation. Therefore, this operation is not repeated after an agent was successfully installed for the first time. Instead, the previous instrumentation instance is returned. However, invoking this method requires synchronization such that subsequently to an installation, getInstrumentation() should be invoked instead.

Returns:
An instrumentation instance representing the currently running JVM.`


@Sam-Kruglov Can you share the JDK version for which you saw successful local runs ?

**Comment by oleksiyp** [comment]
I am considered that something (probably Spring) overrides ByteBuddy. Can you kcheck in final distribution archives what version of ByteBuddy is used

**Comment by Sam-Kruglov** [comment]
@oleksiyp @deeyes24 thanks! 
I am using the same open jdk 11.0.1 locally and it works.
I looked into mockk 1.91 and it has mockk-jvm-agent:1.9.1 which has byte buddy 1.9.3. I forced maven to use this exact version by including this into dependency management:
```
<dependency>
    <groupId>net.bytebuddy</groupId>
    <artifactId>byte-buddy-agent</artifactId>
    <version>1.9.3</version>
</dependency>
<dependency>
    <groupId>net.bytebuddy</groupId>
    <artifactId>byte-buddy</artifactId>
    <version>1.9.3</version>
</dependency>
```
Did not help

**Comment by oleksiyp** [comment]
I know that you forced. I am asking to do additional check in actual packaged files

**Comment by Sam-Kruglov** [comment]
@oleksiyp I am not following, sorry, how do I check that?

**Comment by oleksiyp** [comment]
In your Docker image what version of `byte-buddy-xxx.jar`

**Comment by Sam-Kruglov** [comment]
@oleksiyp where should I look for it? I've built the image

**Comment by oleksiyp** [comment]
inside :smile: open it via https://medium.com/the-code-review/run-bash-or-any-command-in-a-docker-container-9a1e7f0ec204

it may be two situations: one jar or many jars. You need to find them

**Comment by Sam-Kruglov** [comment]
@oleksiyp please bear with me :) I still don't get it. I know that I can execute commands there but where exactly do you want me to look? In `.m2/repository`? In `target` directory? I don't know where else these jars can be. And why is maven dependency tree not enough?

**Comment by oleksiyp** [comment]
You know it is just a habit. Not to trust anything. I am doing so much troubleshooting that something says me that better just check what is indeed getting to the claspath in your docker image.

**Comment by deeyes24** [comment]
HI @Sam-Kruglov : I looked for similar issues and found the below tip.  Please add the below param to your java args and let me know if it works..
-Dnet.bytebuddy.experimental=true

**Comment by Sam-Kruglov** [comment]
@deeyes24 did not help. Note that locally it works on the same docker container with the same maven settings and the same JDK.

**Comment by Sam-Kruglov** [comment]
@oleksiyp I turned off `.m2` mapping so it would be empty and I only saw bytebuddy 1.9.3 downloading in the logs. Hope that's enough to ensure

**Comment by Sam-Kruglov** [comment]
The error comes from surefire and it has `See FAQ web page` part. So I went [there](https://maven.apache.org/surefire/maven-surefire-plugin/faq.html#corruptedstream) and that's what it says:

> If your tests use native library which prints to STDOUT this warning message appears because the library corrupted the channel used by the plugin in order to transmit events with test status back to Maven process. It would be even worse if you override the Java stream by System.setOut because the stream is also supposed to be corrupted but the Maven will never see the tests finished and build may hang. 

>This warning message appears if you use FileDescriptor.out or JVM prints GC summary. 

>In that case the warning is printed "Corrupted STDOUT by directly writing to native stream in forked JVM", and a dump file can be found in Reports directory. 

>If debug level is enabled then messages of corrupted stream appear in the console.

Could that be related to the custom logger that mockk uses?

I enabled debug by including `-X` to the maven build command and it printed a thread dump:

```
[WARNING] Corrupted STDOUT by directly writing to native stream in forked JVM 1. See FAQ web page and the dump file /home/mbuilder/workspace/monese-card-service-temp/card-service/service/target/surefire-reports/2019-02-14T06-16-23_886-jvmRun1.dumpstream
[DEBUG] 2019-02-14 06:17:06
[DEBUG] Full thread dump OpenJDK 64-Bit Server VM (11.0.1+13 mixed mode):
[DEBUG] Threads class SMR info:
[DEBUG] _java_thread_list=0x00007f1c0956ffc0, length=13, elements={
[DEBUG] 0x00007f1c08011000, 0x00007f1c0825c000, 0x00007f1c08260000, 0x00007f1c08274800,
[DEBUG] 0x00007f1c08277000, 0x00007f1c0827e800, 0x00007f1c08280800, 0x00007f1c0830d000,
[DEBUG] 0x00007f1c08317800, 0x00007f1c083ce000, 0x00007f1c0841d000, 0x00007f1bb4267000,
[DEBUG] 0x00007f1c09569800
[DEBUG] }
[DEBUG] "main" #1 prio=5 os_prio=0 cpu=1182.52ms elapsed=2.79s tid=0x00007f1c08011000 nid=0x7b in Object.wait()  [0x00007f1c0fc15000]
[DEBUG]    java.lang.Thread.State: WAITING (on object monitor)
[DEBUG] 	at java.lang.Object.wait(java.base@11.0.1/Native Method)
[DEBUG] 	- waiting on <0x00000007120014d0> (a java.lang.ProcessImpl)
[DEBUG] 	at java.lang.Object.wait(java.base@11.0.1/Object.java:328)
[DEBUG] 	at java.lang.ProcessImpl.waitFor(java.base@11.0.1/ProcessImpl.java:495)
[DEBUG] 	- waiting to re-lock in wait() <0x00000007120014d0> (a java.lang.ProcessImpl)
[DEBUG] 	at net.bytebuddy.agent.ByteBuddyAgent.installExternal(ByteBuddyAgent.java:458)
[DEBUG] 	at net.bytebuddy.agent.ByteBuddyAgent.install(ByteBuddyAgent.java:393)
[DEBUG] 	at net.bytebuddy.agent.ByteBuddyAgent.install(ByteBuddyAgent.java:374)
[DEBUG] 	- locked <0x00000007124ee488> (a java.lang.Class for net.bytebuddy.agent.ByteBuddyAgent)
[DEBUG] 	at net.bytebuddy.agent.ByteBuddyAgent.install(ByteBuddyAgent.java:342)
[DEBUG] 	at net.bytebuddy.agent.ByteBuddyAgent.install(ByteBuddyAgent.java:328)
[DEBUG] 	at io.mockk.proxy.jvm.JvmMockKAgentFactory.initInstrumentation(JvmMockKAgentFactory.kt:134)
[DEBUG] 	at io.mockk.proxy.jvm.JvmMockKAgentFactory.init(JvmMockKAgentFactory.kt:34)
[DEBUG] 	at io.mockk.impl.JvmMockKGateway.<init>(JvmMockKGateway.kt:46)
[DEBUG] 	at io.mockk.impl.JvmMockKGateway.<clinit>(JvmMockKGateway.kt:172)
[DEBUG] 	at com.monese.card.service.service.service.impl.PpsCardServiceImplTest.<init>(PpsCardServiceImplTest.kt:290)
[DEBUG] 	at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(java.base@11.0.1/Native Method)
[DEBUG] 	at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(java.base@11.0.1/NativeConstructorAccessorImpl.java:62)
[DEBUG] 	at jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(java.base@11.0.1/DelegatingConstructorAccessorImpl.java:45)
[DEBUG] 	at java.lang.reflect.Constructor.newInstance(java.base@11.0.1/Constructor.java:490)
[DEBUG] 	at org.junit.platform.commons.util.ReflectionUtils.newInstance(ReflectionUtils.java:443)
[DEBUG] 	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:60)
[DEBUG] 	at org.junit.jupiter.engine.descriptor.ClassTestDescriptor.invokeTestClassConstructor(ClassTestDescriptor.java:342)
[DEBUG] 	at org.junit.jupiter.engine.descriptor.ClassTestDescriptor.instantiateTestClass(ClassTestDescriptor.java:289)
[DEBUG] 	at org.junit.jupiter.engine.descriptor.ClassTestDescriptor.instantiateTestClass(ClassTestDescriptor.java:281)
[DEBUG] 	at org.junit.jupiter.engine.descriptor.ClassTestDescriptor.instantiateAndPostProcessTestInstance(ClassTestDescriptor.java:269)
[DEBUG] 	at org.junit.jupiter.engine.descriptor.ClassTestDescriptor.lambda$testInstanceProvider$2(ClassTestDescriptor.java:259)
[DEBUG] 	at org.junit.jupiter.engine.descriptor.ClassTestDescriptor$$Lambda$216/0x000000080016ec40.getTestInstance(Unknown Source)
[DEBUG] 	at org.junit.jupiter.engine.descriptor.ClassTestDescriptor.lambda$testInstanceProvider$3(ClassTestDescriptor.java:263)
[DEBUG] 	at org.junit.jupiter.engine.descriptor.ClassTestDescriptor$$Lambda$227/0x0000000800171840.get(Unknown Source)
[DEBUG] 	at java.util.Optional.orElseGet(java.base@11.0.1/Optional.java:369)
[DEBUG] 	at org.junit.jupiter.engine.descriptor.ClassTestDescriptor.lambda$testInstanceProvider$4(ClassTestDescriptor.java:262)
[DEBUG] 	at org.junit.jupiter.engine.descriptor.ClassTestDescriptor$$Lambda$217/0x000000080016f040.getTestInstance(Unknown Source)
[DEBUG] 	at org.junit.jupiter.engine.descriptor.NestedClassTestDescriptor.instantiateTestClass(NestedClassTestDescriptor.java:71)
[DEBUG] 	at org.junit.jupiter.engine.descriptor.ClassTestDescriptor.instantiateAndPostProcessTestInstance(ClassTestDescriptor.java:269)
[DEBUG] 	at org.junit.jupiter.engine.descriptor.ClassTestDescriptor.lambda$testInstanceProvider$2(ClassTestDescriptor.java:259)
[DEBUG] 	at org.junit.jupiter.engine.descriptor.ClassTestDescriptor$$Lambda$216/0x000000080016ec40.getTestInstance(Unknown Source)
[DEBUG] 	at org.junit.jupiter.engine.descriptor.ClassTestDescriptor.lambda$testInstanceProvider$3(ClassTestDescriptor.java:263)
[DEBUG] 	at org.junit.jupiter.engine.descriptor.ClassTestDescriptor$$Lambda$227/0x0000000800171840.get(Unknown Source)
[DEBUG] 	at java.util.Optional.orElseGet(java.base@11.0.1/Optional.java:369)
[DEBUG] 	at org.junit.jupiter.engine.descriptor.ClassTestDescriptor.lambda$testInstanceProvider$4(ClassTestDescriptor.java:262)
[DEBUG] 	at org.junit.jupiter.engine.descriptor.ClassTestDescriptor$$Lambda$217/0x000000080016f040.getTestInstance(Unknown Source)
[DEBUG] 	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.prepare(TestMethodTestDescriptor.java:82)
[DEBUG] 	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.prepare(TestMethodTestDescriptor.java:59)
[DEBUG] 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$prepare$0(NodeTestTask.java:83)
[DEBUG] 	at org.junit.platform.engine.support.hierarchical.NodeTestTask$$Lambda$173/0x0000000800159c40.execute(Unknown Source)
[DEBUG] 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:72)
[DEBUG] 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.prepare(NodeTestTask.java:83)
[DEBUG] 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:69)
[DEBUG] 	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService$$Lambda$190/0x0000000800168440.accept(Unknown Source)
[DEBUG] 	at java.util.ArrayList.forEach(java.base@11.0.1/ArrayList.java:1540)
[DEBUG] 	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
[DEBUG] 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$4(NodeTestTask.java:112)
[DEBUG] 	at org.junit.platform.engine.support.hierarchical.NodeTestTask$$Lambda$186/0x000000080015dc40.execute(Unknown Source)
[DEBUG] 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:72)
[DEBUG] 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:98)
[DEBUG] 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:74)
[DEBUG] 	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService$$Lambda$190/0x0000000800168440.accept(Unknown Source)
[DEBUG] 	at java.util.ArrayList.forEach(java.base@11.0.1/ArrayList.java:1540)
[DEBUG] 	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
[DEBUG] 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$4(NodeTestTask.java:112)
[DEBUG] 	at org.junit.platform.engine.support.hierarchical.NodeTestTask$$Lambda$186/0x000000080015dc40.execute(Unknown Source)
[DEBUG] 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:72)
[DEBUG] 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:98)
[DEBUG] 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:74)
[DEBUG] 	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService$$Lambda$190/0x0000000800168440.accept(Unknown Source)
[DEBUG] 	at java.util.ArrayList.forEach(java.base@11.0.1/ArrayList.java:1540)
[DEBUG] 	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
[DEBUG] 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$4(NodeTestTask.java:112)
[DEBUG] 	at org.junit.platform.engine.support.hierarchical.NodeTestTask$$Lambda$186/0x000000080015dc40.execute(Unknown Source)
[DEBUG] 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:72)
[DEBUG] 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:98)
[DEBUG] 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:74)
[DEBUG] 	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:32)
[DEBUG] 	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
[DEBUG] 	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:51)
[DEBUG] 	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:220)
[DEBUG] 	at org.junit.platform.launcher.core.DefaultLauncher.lambda$execute$6(DefaultLauncher.java:188)
[DEBUG] 	at org.junit.platform.launcher.core.DefaultLauncher$$Lambda$161/0x000000080012ac40.accept(Unknown Source)
[DEBUG] 	at org.junit.platform.launcher.core.DefaultLauncher.withInterceptedStreams(DefaultLauncher.java:202)
[DEBUG] 	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:181)
[DEBUG] 	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:128)
[DEBUG] 	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:142)
[DEBUG] 	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:117)
[DEBUG] 	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
[DEBUG] 	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
[DEBUG] 	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
[DEBUG] 	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
[DEBUG] "Reference Handler" #2 daemon prio=10 os_prio=0 cpu=0.55ms elapsed=2.76s tid=0x00007f1c0825c000 nid=0x82 waiting on condition  [0x00007f1be07ec000]
[DEBUG]    java.lang.Thread.State: RUNNABLE
[DEBUG] 	at java.lang.ref.Reference.waitForReferencePendingList(java.base@11.0.1/Native Method)
[DEBUG] 	at java.lang.ref.Reference.processPendingReferences(java.base@11.0.1/Reference.java:241)
[DEBUG] 	at java.lang.ref.Reference$ReferenceHandler.run(java.base@11.0.1/Reference.java:213)
[DEBUG] "Finalizer" #3 daemon prio=8 os_prio=0 cpu=0.48ms elapsed=2.76s tid=0x00007f1c08260000 nid=0x83 in Object.wait()  [0x00007f1be06eb000]
[DEBUG]    java.lang.Thread.State: WAITING (on object monitor)
[DEBUG] 	at java.lang.Object.wait(java.base@11.0.1/Native Method)
[DEBUG] 	- waiting on <0x0000000705badcb8> (a java.lang.ref.ReferenceQueue$Lock)
[DEBUG] 	at java.lang.ref.ReferenceQueue.remove(java.base@11.0.1/ReferenceQueue.java:155)
[DEBUG] 	- waiting to re-lock in wait() <0x0000000705badcb8> (a java.lang.ref.ReferenceQueue$Lock)
[DEBUG] 	at java.lang.ref.ReferenceQueue.remove(java.base@11.0.1/ReferenceQueue.java:176)
[DEBUG] 	at java.lang.ref.Finalizer$FinalizerThread.run(java.base@11.0.1/Finalizer.java:170)
[DEBUG] "Signal Dispatcher" #4 daemon prio=9 os_prio=0 cpu=0.14ms elapsed=2.75s tid=0x00007f1c08274800 nid=0x84 waiting on condition  [0x0000000000000000]
[DEBUG]    java.lang.Thread.State: RUNNABLE
[DEBUG] "C2 CompilerThread0" #5 daemon prio=9 os_prio=0 cpu=1383.86ms elapsed=2.75s tid=0x00007f1c08277000 nid=0x85 waiting on condition  [0x0000000000000000]
[DEBUG]    java.lang.Thread.State: RUNNABLE
[DEBUG]    No compile task
[DEBUG] "C1 CompilerThread0" #7 daemon prio=9 os_prio=0 cpu=605.05ms elapsed=2.75s tid=0x00007f1c0827e800 nid=0x86 waiting on condition  [0x0000000000000000]
[DEBUG]    java.lang.Thread.State: RUNNABLE
[DEBUG]    No compile task
[DEBUG] "Sweeper thread" #8 daemon prio=9 os_prio=0 cpu=12.33ms elapsed=2.75s tid=0x00007f1c08280800 nid=0x87 runnable  [0x0000000000000000]
[DEBUG]    java.lang.Thread.State: RUNNABLE
[DEBUG] "Service Thread" #9 daemon prio=9 os_prio=0 cpu=0.08ms elapsed=2.72s tid=0x00007f1c0830d000 nid=0x88 runnable  [0x0000000000000000]
[DEBUG]    java.lang.Thread.State: RUNNABLE
[DEBUG] "Common-Cleaner" #10 daemon prio=8 os_prio=0 cpu=1.20ms elapsed=2.71s tid=0x00007f1c08317800 nid=0x8a in Object.wait()  [0x00007f1bb3efc000]
[DEBUG]    java.lang.Thread.State: TIMED_WAITING (on object monitor)
[DEBUG] 	at java.lang.Object.wait(java.base@11.0.1/Native Method)
[DEBUG] 	- waiting on <0x0000000705a64500> (a java.lang.ref.ReferenceQueue$Lock)
[DEBUG] 	at java.lang.ref.ReferenceQueue.remove(java.base@11.0.1/ReferenceQueue.java:155)
[DEBUG] 	- waiting to re-lock in wait() <0x0000000705a64500> (a java.lang.ref.ReferenceQueue$Lock)
[DEBUG] 	at jdk.internal.ref.CleanerImpl.run(java.base@11.0.1/CleanerImpl.java:148)
[DEBUG] 	at java.lang.Thread.run(java.base@11.0.1/Thread.java:834)
[DEBUG] 	at jdk.internal.misc.InnocuousThread.run(java.base@11.0.1/InnocuousThread.java:134)
[DEBUG] "surefire-forkedjvm-command-thread" #11 daemon prio=5 os_prio=0 cpu=3.07ms elapsed=2.67s tid=0x00007f1c083ce000 nid=0x8b runnable  [0x00007f1bb39d3000]
[DEBUG]    java.lang.Thread.State: RUNNABLE
[DEBUG] 	at java.io.FileInputStream.readBytes(java.base@11.0.1/Native Method)
[DEBUG] 	at java.io.FileInputStream.read(java.base@11.0.1/FileInputStream.java:279)
[DEBUG] 	at java.io.BufferedInputStream.fill(java.base@11.0.1/BufferedInputStream.java:252)
[DEBUG] 	at java.io.BufferedInputStream.read(java.base@11.0.1/BufferedInputStream.java:271)
[DEBUG] 	- locked <0x0000000705bb1180> (a java.io.BufferedInputStream)
[DEBUG] 	at java.io.DataInputStream.readInt(java.base@11.0.1/DataInputStream.java:392)
[DEBUG] 	at org.apache.maven.surefire.booter.MasterProcessCommand.decode(MasterProcessCommand.java:115)
[DEBUG] 	at org.apache.maven.surefire.booter.CommandReader$CommandRunnable.run(CommandReader.java:391)
[DEBUG] 	at java.lang.Thread.run(java.base@11.0.1/Thread.java:834)
[DEBUG] "surefire-forkedjvm-ping-30s" #12 daemon prio=5 os_prio=0 cpu=2.26ms elapsed=2.60s tid=0x00007f1c0841d000 nid=0x8c waiting on condition  [0x00007f1bb34c7000]
[DEBUG]    java.lang.Thread.State: TIMED_WAITING (parking)
[DEBUG] 	at jdk.internal.misc.Unsafe.park(java.base@11.0.1/Native Method)
[DEBUG] 	- parking to wait for  <0x0000000705ba2560> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
[DEBUG] 	at java.util.concurrent.locks.LockSupport.parkNanos(java.base@11.0.1/LockSupport.java:234)
[DEBUG] 	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.awaitNanos(java.base@11.0.1/AbstractQueuedSynchronizer.java:2123)
[DEBUG] 	at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(java.base@11.0.1/ScheduledThreadPoolExecutor.java:1182)
[DEBUG] 	at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(java.base@11.0.1/ScheduledThreadPoolExecutor.java:899)
[DEBUG] 	at java.util.concurrent.ThreadPoolExecutor.getTask(java.base@11.0.1/ThreadPoolExecutor.java:1054)
[DEBUG] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(java.base@11.0.1/ThreadPoolExecutor.java:1114)
[DEBUG] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(java.base@11.0.1/ThreadPoolExecutor.java:628)
[DEBUG] 	at java.lang.Thread.run(java.base@11.0.1/Thread.java:834)
[DEBUG] "C2 CompilerThread1" #6 daemon prio=9 os_prio=0 cpu=1233.04ms elapsed=1.83s tid=0x00007f1bb4267000 nid=0x90 waiting on condition  [0x0000000000000000]
[DEBUG]    java.lang.Thread.State: RUNNABLE
[DEBUG]    No compile task
[DEBUG] "process reaper" #14 daemon prio=10 os_prio=0 cpu=0.11ms elapsed=0.24s tid=0x00007f1c09569800 nid=0x93 runnable  [0x00007f1c0c077000]
[DEBUG]    java.lang.Thread.State: RUNNABLE
[DEBUG] 	at java.lang.ProcessHandleImpl.waitForProcessExit0(java.base@11.0.1/Native Method)
[DEBUG] 	at java.lang.ProcessHandleImpl$1.run(java.base@11.0.1/ProcessHandleImpl.java:138)
[DEBUG] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(java.base@11.0.1/ThreadPoolExecutor.java:1128)
[DEBUG] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(java.base@11.0.1/ThreadPoolExecutor.java:628)
[DEBUG] 	at java.lang.Thread.run(java.base@11.0.1/Thread.java:834)
[DEBUG] "VM Thread" os_prio=0 cpu=11.21ms elapsed=2.77s tid=0x00007f1c08254000 nid=0x81 runnable  
[DEBUG] "GC Thread#0" os_prio=0 cpu=14.11ms elapsed=2.79s tid=0x00007f1c08038800 nid=0x7c runnable  
[DEBUG] "GC Thread#1" os_prio=0 cpu=9.23ms elapsed=1.99s tid=0x00007f1bd0001000 nid=0x8d runnable  
[DEBUG] "GC Thread#2" os_prio=0 cpu=9.81ms elapsed=1.99s tid=0x00007f1bd0002800 nid=0x8e runnable  
[DEBUG] "GC Thread#3" os_prio=0 cpu=5.40ms elapsed=1.99s tid=0x00007f1bd0004000 nid=0x8f runnable  
[DEBUG] "G1 Main Marker" os_prio=0 cpu=0.51ms elapsed=2.79s tid=0x00007f1c08096800 nid=0x7d runnable  
[DEBUG] "G1 Conc#0" os_prio=0 cpu=0.07ms elapsed=2.78s tid=0x00007f1c08098800 nid=0x7e runnable  
[DEBUG] "G1 Refine#0" os_prio=0 cpu=1.10ms elapsed=2.78s tid=0x00007f1c08189800 nid=0x7f runnable  
[DEBUG] "G1 Young RemSet Sampling" os_prio=0 cpu=0.82ms elapsed=2.78s tid=0x00007f1c0818b000 nid=0x80 runnable  
[DEBUG] "VM Periodic Task Thread" os_prio=0 cpu=1.38ms elapsed=2.72s tid=0x00007f1c0830f800 nid=0x89 waiting on condition  
[DEBUG] JNI global refs: 8, weak refs: 0
[DEBUG] Heap
[DEBUG]  garbage-first heap   total 258048K, used 54943K [0x0000000705400000, 0x0000000800000000)
[DEBUG]   region size 1024K, 48 young (49152K), 2 survivors (2048K)
[DEBUG]  Metaspace       used 16388K, capacity 16821K, committed 17152K, reserved 1064960K
[DEBUG]   class space    used 1629K, capacity 1788K, committed 1792K, reserved 1048576K
```

**Comment by oleksiyp** [comment]
MockK uses either SLF4J or Java util logging. Try adding dependency to logback

**Comment by Sam-Kruglov** [comment]
I was able to narrow the problem down to `net.bytebuddy.agent.ByteBuddyAgent.install()`. Closing in favor of https://github.com/raphw/byte-buddy/issues/612. Thanks for your time!

**Comment by Sam-Kruglov** [comment]
This line never returns null, rather it just throws `IllegalStateException`
https://github.com/mockk/mockk/blob/061349efbc558ac459ed8099a98fc27723873c0a/agent/jvm/src/main/kotlin/io/mockk/proxy/jvm/JvmMockKAgentFactory.kt#L134

So this one should probably be changed to try catch
https://github.com/mockk/mockk/blob/061349efbc558ac459ed8099a98fc27723873c0a/agent/jvm/src/main/kotlin/io/mockk/proxy/jvm/JvmMockKAgentFactory.kt#L136

**Comment by oleksiyp** [comment]
@Sam-Kruglov thanks for pointing this out!!!

**Comment by Sam-Kruglov** [comment]
@oleksiyp I also noted that I don't have any problems with Mockito. 
I saw the only usage of `ByteBuddyAgent.install()` in this class
https://github.com/mockito/mockito/blob/3108ff10cd26fa49580df255b3eefb2edb53d16f/src/main/java/org/mockito/internal/creation/bytebuddy/InlineByteBuddyMockMaker.java#L92

But when I actually create a mock, another class is used (I guess by default) which creates mocks somehow without calling that `intall` function. So my test passes.

This is the class they use
https://github.com/mockito/mockito/blob/4f72147c464c1a8a642d01fc3334e98e92b464cd/src/main/java/org/mockito/internal/creation/bytebuddy/SubclassByteBuddyMockMaker.java#L28

**Comment by oleksiyp** [comment]
MockK uses a bit different way(inlining) then Mockito uses by default(subclassing) to overcome `final classes` problem.

Try switching mockito to inlining mode https://hadihariri.com/2016/10/04/Mocking-Kotlin-With-Mockito/#mockmaker. It probably will have same thing.

Anyway I am reopening ticket. Just want let you know that I check such things during weekends(and even not during each weekend). So please be patient

**Comment by oleksiyp** [comment]
@Sam-Kruglov is your image public? Can you provide it so I can spawn same thing in AWS?

**Comment by Sam-Kruglov** [comment]
@oleksiyp so the code could just contain a single test with `mockk<Serializable>()` and the docker image is the following:

**Comment by Sam-Kruglov** [comment]
btw https://github.com/raphw/byte-buddy/issues/612 is resolved, my CI runs correctly now

**Comment by sevar83** [comment]
I've received this error:

`java.lang.NoClassDefFoundError: Could not initialize class io.mockk.impl.JvmMockKGateway`

after upgrading to AS Arctic Fox
using mockk 1.10.2

**Comment by vistiyos** [comment]
@sevar83 did you manage to solve it?
